### ISSUE TITLE
[FEAT] FontAwesomeTestingModule is missing

### ISSUE THREAD

**Issue by damienwebdev** [FortAwesome/angular-fontawesome#134]
### Describe the problem you'd like to see solved or task you'd like to see made easier

When testing components that use the FA icon, you either have to re-import the font awesome library for each component that utilizes it (painful) or use the CUSTOM_ELEMENTS_SCHEMA, which is bad practice. This project should provide a mocked `fa-icon` component that is very "dumb" and does nothing.

### Is this in relation to an existing part of angular-fontawesome or something new?
Something new.

### What is 1 thing that we can do when building this feature that will guarantee that it is awesome?

Make sure that the footprint of the testing module is tiny.

### Why would other angular-fontawesome users care about this?

Because testing is a good thing.

### On a scale of 1 (sometime in the future) to 10 (absolutely right now), how soon would you recommend we make this feature?

10. Testing should be done early and often.

### Feature request checklist

- [x] This is a single feature (i.e. not a re-write of all of Font Awesome)
- [x] The title starts with "Feature request: " and is followed by a clear feature name (Ex: `Feature request: moar cowbell`)
- [x] I have [searched for existing issues](https://github.com/FortAwesome/angular-fontawesome/issues) and to the best of my knowledge this is not a duplicate

**Comment by devoto13** [comment]
I do agree that we should throw when specified icon does not exist. This is a clear error and should crash and not just log a warning. Need to change [this](https://github.com/FortAwesome/angular-fontawesome/blob/master/src/lib/shared/errors/warn-if-icon-html-missing.ts#L5) to throw instead of logging.

I'm not so sure about the module for testing though. Do you know of any other libraries which provide a testing modules with dummy components instead of real ones? IMO having testing module with dummy components is not much different from using `CUSTOM_ELEMENTS_SCHEMA` - essentially saying, I don't care about these elements in my tests.

**Comment by damienwebdev** [comment]
@devoto13 Do you have an example of

> Also, library is global, so you don't have re-import it in every test file, but populate it once per test run (same as you probably do for runtime in AppModule).

If I'm not mistaken, Angular component tests are all atomic. There's no "global" module which they all inherit from, each instance of:

```
TestBed.configureTestingModule({
      imports: [],
      declarations: [
        ComponentUnderTest
      ]
    }).compileComponents();
```

would warrant an import like:

```
TestBed.configureTestingModule({
      imports: [
        FontAwesomeModule
      ],
      declarations: [
        ComponentUnderTest
      ]
    }).compileComponents();
```

**Comment by devoto13** [comment]
You're right that tests are atomic, so you'll need to import `FontAwesomeModule` in every test anyway. But so you'll need to import `FontAwesomeTestingModule` if we introduce it. So nothing really changes.

By library I meant [Icon Library](https://github.com/FortAwesome/angular-fontawesome#using-the-icon-library) (sorry if it was not clear), which is global, so you don't need to add your icons in every test, but can do it once in the tests entrypoint instead (i.e. call `library.add(...)` once in `src/test.ts`).

**Comment by damienwebdev** [comment]
@devoto13 Just to circle back on this.

I think a FontAwesomeTestingModule would be nice, because it would allow me to test the bindings that I've set on the `fa-icon`. If I were to use CUSTOM_ELEMENTS_SCHEMA, I would not only not be able to test bindings, but also prevent testing of other subcomponents in my component.

The `FontAwesomeTestingModule` could provide a simple version of `fa-icon` that skips errors like the aformentioned, but still provides the Inputs and Outputs so that we could simply test our usage of the component.

**Comment by devoto13** [comment]
> If I were to use CUSTOM_ELEMENTS_SCHEMA, I would not only not be able to test bindings, but I'd also prevent testing of other subcomponents in my component, which I wouldn't necessarily want.

Ok, I'm convinced now. Let's have a `FontAwesomeTestingModule`. PRs are welcome!

**Comment by DaSchTour** [comment]
I would suggest to just use `MockModule` from [ng-mocks](https://www.npmjs.com/package/ng-mocks)

**Comment by damienwebdev** [comment]
I don't see a need for yet another dependency in the tree. We have a good opportunity here to provide a decent testing scaffold that is under this repo's control. I'd personally rather not be beholden to how `ng-mocks` works in order to save a few hours of development effort.

**Comment by DaSchTour** [comment]
Well I would not add ng-mocks as a dependency to this module but just refer to ng-mocks for testing. For bigger angular applications ng-mocks is a must have. So telling people to use ng-mocks would be just enough from my point of view. So there will be no need to change or add any code but just add documentation.
